### ISSUE TITLE
gateway throws different error messages for invalid --address argument

### ISSUE THREAD

**Issue by krisis** [minio/minio#4503]
Gateway with invalid --address argument throws different failure messages in different runtime environments.

### Docker
```
➜  minio git:(master) ✗ docker run --rm -e MINIO_ACCESS_KEY=minio -e MINIO_SECRET_KEY=minio123 -p 9001:9001 minio/minio:edge gateway --address 9001 s3 http://localhost:9000
Waiting for all hosts to resolve....All hosts are resolving proceeding to initialize Minio.
time="2017-06-08T09:36:40Z" level=fatal msg="Unable to compare server and endpoint addresses." cause="lookup 9001: invalid domain name" source="[gateway-main.go:216:gatewayMain()]" 
```

### Host Operating System
```
➜  minio git:(master) ✗ minio gateway --address 9001 s3 http://localhost:9000
FATA[0000] Failed to start minio server                  cause=missing port in address 9001 source=[gateway-main.go:289:gatewayMain.func1()]
```
## Expected Behavior

gateway should fail with the same error message for the same (class of) invalid argument supplied.
## Current Behavior

See issue description for more details.
## Possible Solution

Check for port in --address argument.
 
## Steps to Reproduce (for bugs)

1. See description for more details
2.
3.
4.

## Your Environment

* Version used (`minio version`):
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`):
* Link to your project:

**Comment by krisis** [comment]
@deekoder I am working on this.

**Comment by harshavardhana** [comment]
Fixed by #4376
